scraper/src/youtube2zim/scraper.py

```diff
@@ -1108,3 +1120,240 @@ def to_data_js(video):
 
         # clean videos left out in videos directory
         remove_unused_videos(videos)
+
+    def make_json_files(self, actual_videos_ids):
+        """Generate JSON files to be consumed by the frontend"""
+
+        def remove_unused_videos(videos):
+            video_ids = [video["contentDetails"]["videoId"] for video in videos]
+            for path in self.videos_dir.iterdir():
+                if path.is_dir() and path.name not in video_ids:
+                    logger.debug(f"Removing unused video {path.name}")
+                    shutil.rmtree(path, ignore_errors=True)
+
+        def is_present(video):
+            """whether this video has actually been succeffuly downloaded"""
+            return video["contentDetails"]["videoId"] in actual_videos_ids
+
+        def video_has_channel(videos_channels, video):
+            return video["contentDetails"]["videoId"] in videos_channels
+
+        def get_thumbnail_path(video_id):
+            return f"videos/{video_id}/video.webp"
+
+        def get_subtitles(video_id):
+            video_dir = self.videos_dir.joinpath(video_id)
+            languages = [
+                x.stem.split(".")[1]
+                for x in video_dir.iterdir()
+                if x.is_file() and x.name.endswith(".vtt")
+            ]
+
+            def to_subtitle_object(lang):
+                try:
+                    try:
+                        subtitle = get_language_details(
+                            YOUTUBE_LANG_MAP.get(lang, lang)
+                        )
+                    except NotFound:
+                        lang_simpl = re.sub(r"^([a-z]{2})-.+$", r"\1", lang)
+                        subtitle = get_language_details(
+                            YOUTUBE_LANG_MAP.get(lang_simpl, lang_simpl)
+                        )
+                except Exception:
+                    logger.error(f"Failed to get language details for {lang}")
+                    raise
+                return Subtitle(
+                    code=lang,
+                    name=f"{subtitle['english'].title()} - {subtitle['query']}",
+                )
+
+            # Youtube.com sorts subtitles by English name
+            return sorted(map(to_subtitle_object, languages), key=lambda x: x.name)
+
+        def get_videos_list(playlist):
+            videos = load_mandatory_json(
+                self.cache_dir, f"playlist_{playlist.playlist_id}_videos"
+            )
+            videos = list(filter(skip_deleted_videos, videos))
+            videos = list(filter(is_present, videos))
+            videos = list(filter(has_channel, videos))
+            videos = sorted(videos, key=lambda v: v["snippet"]["position"])
+            return videos
+
+        def generate_video_object(video):
+            video_id = video["contentDetails"]["videoId"]
+            title = video["snippet"]["title"]
```

Comment:
title, description and publication_date are used only once, could be used directly in object, isn't it?